src/polymer/declaration-property-handlers.ts

```diff
@@ -71,11 +72,23 @@ export function declarationPropertyHandlers(
     },
     listeners(node: estree.Node) {
       if (node.type !== 'ObjectExpression') {
+        declaration.warnings.push({
+          code: 'invalid-listeners-declaration',
+          message: '`listeners` property should be an object expression',
+          severity: Severity.ERROR,
+          sourceRange: document.sourceRangeForNode(node)!
+        });
         return;
       }
       for (let prop of node.properties) {
         if (prop.key.type !== 'Literal' || typeof prop.key.value !== 'string' ||
             prop.value.type !== 'Literal' || typeof prop.value.value !== 'string') {
+          declaration.warnings.push({
+            code: 'invalid-listeners-declaration',
+            message: 'handler value should be a string',
+            severity: Severity.ERROR,
+            sourceRange: document.sourceRangeForNode(prop.value)!
+          });
```

Comment:
Ah, yeah, identifiers should be allowed for keys. I thought that that syntax was recognized as a string literal (as semantically it's a string literal, just one that's parsed with identifier syntax).

And yeah, you're right, we should probably only generate warnings by default on stuff that's definitely wrong, rather than stuff that's just impossible to statically analyze. (As an eventual goal it would be nice if people could opt in to warnings on non-statically analyzable code, as some people would want to enforce that in their codebases, but that will require more advanced Warning-twiddling, as well as an IGNORE severity, which both are out of scope of this change.)

So I guess the right thing to do here is to handle string literals or identifiers for keys, and only string literals for values, and if it's anything else just continue without emitting a warning.